src/cursor/common.rs

```diff
@@ -73,12 +73,13 @@ impl<T: GetMoreProvider> Stream for GenericCursor<T> {
             if let Some(future) = self.provider.executing_future() {
                 match Pin::new(future).poll(cx) {
                     // If a result is ready, retrieve the buffer and update the exhausted status.
-                    Poll::Ready(mut get_more_result) => {
-                        let buffer_result = get_more_result.take_buffer();
-                        self.exhausted = get_more_result.exhausted();
+                    Poll::Ready(get_more_result) => {
+                        let exhausted = get_more_result.exhausted();
```

Comment:
Previously, the GetMoreResult trait allowed for getting the exhausted state and the buffer separately, with the latter cloning the error if it existed. This clone turned out not to be necessary, as it was only done to preserve the ownership of the GetMoreResult so it could be passed wholesale to the `clear_execution` method on the provider. This method needs to know whether the cursor is exhausted and potentially have access to the session used for the cursor, so the simplest fix was just to add a method on GetMoreResult that splits it into the session (with the existing associated type for the GetMoreResult being returned) and the buffer (or owned error if it exists). GetMoreResult is internal to the cursor module, so these changes were possible without changing anything in the rest of the driver or any cursor semantics.